commandapi-platforms/commandapi-spigot/commandapi-spigot-nms/commandapi-spigot-nms-common/src/main/java/dev/jorel/commandapi/nms/SpigotNMS_Common.java

```diff
@@ -0,0 +1,28 @@
+package dev.jorel.commandapi.nms;
+
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import dev.jorel.commandapi.CommandAPISpigot;
+import net.md_5.bungee.api.chat.BaseComponent;
+import net.minecraft.commands.CommandSourceStack;
+import net.minecraft.commands.arguments.ColorArgument;
+import org.bukkit.ChatColor;
+
+public abstract class SpigotNMS_Common extends CommandAPISpigot<CommandSourceStack> {
+
+	@Override
+	public abstract BaseComponent[] getChat(CommandContext<CommandSourceStack> cmdCtx, String key) throws CommandSyntaxException;
```

Comment:
`getChat` (and `getChatComponent`) could be implemented here, as long as they are `@Overriden` in 1.20.5+, since `Serializer#toJson` needs the `COMMAND_BUILD_CONTEXT` then. 

[`NMS_Common` on `dev/dev`](https://github.com/JorelAli/CommandAPI/blob/c3ff409407ba18f3b0561051bdc99b1f89ca1678/commandapi-platforms/commandapi-bukkit/commandapi-bukkit-nms/commandapi-bukkit-nms-common/src/main/java/dev/jorel/commandapi/nms/NMS_Common.java#L348-L352) and `dev/paper/restructure-experiment` currently [does this](https://github.com/JorelAli/CommandAPI/blob/d9c1f0d6a85cd94ab7d2d20682c12ad20684ea54/commandapi-platforms/commandapi-spigot/commandapi-spigot-nms/commandapi-spigot-nms-common/src/main/java/dev/jorel/commandapi/nms/SpigotNMS_Common.java#L16-L33).